catalog/content/core/content-core-catalogerplugin/src/main/java/ddf/content/plugin/cataloger/Cataloger.java

```diff
@@ -197,4 +250,18 @@ public String deleteMetacard(String productUri) throws PluginExecutionException
         return deletedCatalogId;
     }
 
+    public String runCallableWithSubject(Callable callable) throws Exception {
```

Comment:
I think this should be provided by the security app.  A content framework plugin should not have to know about Shiro or its ThreadContext/SecurityManager.